charm4py/charmlib/charmlib_ctypes.py

```diff
@@ -609,9 +611,15 @@ def init(self, libcharm_path):
       p = os.environ.get('LIBCHARM_PATH')
       if p is not None: libcharm_path = p
       if libcharm_path is not None:
-        self.lib = ctypes.CDLL(os.path.join(libcharm_path, 'libcharm.so'))
+        if self.system == 'darwin':
```

Comment:
You may be able to get the extension from distutils, something like `distutils.ccompiler.new_compiler().shared_lib_extension`, which could simplify this code. lot.

Reply:
there's no shared_lib_extension attribute in distutils

Reply:
It seems to work for me:

```
>>> import distutils.ccompiler
>>> distutils.ccompiler.new_compiler().shared_lib_extension
'.so'
```

Reply:
This seems to return `.so` even on darwin, so probably not helpful here.